code/operating_system/src/scheduling/priority_scheduling/priority_scheduling_preemptive.c

```diff
@@ -0,0 +1,99 @@
+//Priority scheduling: each process is assigned a priority. Process with highest priority is to be executed first and so on.
+#include<stdio.h>
+#include<stdlib.h>
+
+struct process
+{
+    int pname, pr, at, bt;
+    int ct, wt, rt, tat, flag; //flag=0 means unfinished process flag=1 process is finished
+};
+typedef struct process process;
+
+int t = 0, n;
+process pro[20];
+float act, awt, atat, art;
```

Comment:
All these variables should be local and passed around as parameters, not globals.